### ISSUE TITLE
False positive: eslint-config-prettier

### ISSUE THREAD

**Issue by tinymachine** [depcheck/depcheck#330]
### Bug Description
This might be too much to expect depcheck to handle, but eslint-config-prettier is used by eslint-plugin-prettier when the ESLint config includes`{ "extends": ["plugin:prettier/recommended"] }`, even though [eslint-plugin-prettier's `package.json`](https://github.com/prettier/eslint-plugin-prettier/blob/34b39dec2e6e283da1ca6faa0c636c5361efb5b9/package.json#L30-L36) only includes eslint-config-prettier as a dev dependency. (See [Prettier's documentation on integrating with ESLint](https://prettier.io/docs/en/eslint.html#use-both).) When running `depcheck --specials=eslint`, though, eslint-config-prettier is listed as an unused dev dependency, though it is actually a dev dependency.

(This seems different than #316 where the issue was eslintConfig not being honored in the package.json.)

### Code snippets
Where the dependency is used:

```js
// .eslintrc.js
module.exports = {
  "extends": [
    "eslint:recommended",
    "plugin:prettier/recommended"
  ],
  "plugins": [
    "prettier"
  ],
  "env": {
    "browser": true,
    "node": true
  }
}
```

Where the dependency is listed in package.json:

```js
"devDependencies": {
    "@babel/core": "^7.3.4",
    "@babel/preset-env": "^7.3.4",
    "autoprefixer": "^6.7.7",
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.1.0",
    "eslint-plugin-prettier": "^3.0.1",
    "grunt": "^1.0.3",
    "grunt-babel": "^8.0.0",
    "grunt-contrib-nodeunit": "~0.4.1",
    "grunt-contrib-uglify": "^4.0.0",
    "grunt-contrib-watch": "^0.6.1",
    "grunt-reload": "^0.2.0",
    "prettier": "^1.16.4"
}
```

### Versions
 - `node -v`: v8.12.0
 - `npm -v`: 6.4.1
 - `depcheck --version`: 0.7.2

### Extra info
Here's the output of `depcheck --json --specials=eslint` (with the `"using"` paths obscured):
```js
{"dependencies":[],"devDependencies":["@babel/preset-env","autoprefixer","eslint-config-prettier","grunt-contrib-nodeunit","grunt-reload"],"missing":{},"using":{"eslint-plugin-prettier":[".../.eslintrc.js"],"eslint":[".../.eslintrc.js"],"prettier":[".../.eslintrc.js"],"grunt-babel":[".../gruntfile.js"],"grunt-contrib-watch":[".../gruntfile.js"],"grunt-contrib-uglify":[".../gruntfile.js"],"@babel/core":[".../gruntfile.js"],"grunt":[".../gruntfile.js"]},"invalidFiles":{},"invalidDirs":{}}
```

Thanks for any help you can provide, and for your work on this package.

**Comment by VincentLanglet** [comment]
This is still a bug @rumpl, I can't add prettier because of it (https://github.com/depcheck/depcheck/pull/463)
